Stratis.Bitcoin/BlockPulling/DownloadAssignmentStrategy.cs

```diff
@@ -0,0 +1,93 @@
+﻿using NBitcoin.Protocol;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using System.Collections.Concurrent;
+using NBitcoin.Protocol.Behaviors;
+using System.Threading;
+using Stratis.Bitcoin.Connection;
+
+namespace Stratis.Bitcoin.BlockPulling
+{
+    /// <summary>
+    /// Implements a strategy for a block puller that needs to download 
+    /// a set of blocks and from its connected peers.
+    /// </summary>
+    public static class DownloadAssignmentStrategy
+    {
+        /// <summary>Information about a connected peer.</summary>
+        public class PeerInformation
+        {
+            /// <summary>Evaluation of the node's past experience with the peer.</summary>
+            public int QualityScore;
```

Comment:
any reason fields are used instead of properties? 
If not, please always use properties rather than public fields.